apps/src/StudioApp.js

```diff
@@ -218,7 +218,8 @@ StudioApp.prototype.configure = function (options) {
   // NOTE: editCode (which currently implies droplet) and usingBlockly_ are
   // currently mutually exclusive.
   this.editCode = options.level && options.level.editCode;
-  this.usingBlockly_ = !this.editCode;
+  this.scratch = options.level && options.level.scratch;
+  this.usingBlockly_ = !this.editCode && !this.scratch;
```

Comment:
Would still like to see this addressed. It seems like we're adding a new editor, but in a way that is confusing to me. In the case of scratch presumably `this.scratch === true;` and `this.usingBlockly_ === false`, however we *are* using Blockly, just a different version of it.

This gets particularly confusing starting on line 675 when we start calling Blockly methods after having passed a conditional (line 650) that decided we are not using Blockly.

I'm not sure exactly what the right cleanup here is. Perhaps it could be partly addressed by differentiating Blockly (code-dot-org) and Blockly (scratch). It also seems like there's value in general in cleaning up how we track our editor type now that we've expanded it from an either/or to a tristate (conceivably this could expand even further in the future).